url2a908001cd_malictus_klang_tgz-pJ8-malictus_klang_test_PrimitiveTestJ8/src/malictus/klang/chunk/FMTChunk.java

```
@@ -0,0 +1,303 @@
+		int total = 0;
+		counter = 0;
+		int end = primitives.size();
+		if (isBasic) {
+			end = 6;
+		}
+		while (counter < end) {
+			Primitive p = primitives.get(counter).getPrimitive();
+			if (!(p instanceof PrimitiveFixedByte)) {
+				throw new IOException(KlangConstants.ERROR_NOT_FIXED_BYTE);
+			}
+			total = total + ((PrimitiveFixedByte)p).getFixedLength();
+			counter = counter + 1;
+		}
+		if (!isBasic) {
+			//2 size bytes, plus additional 14 bytes at the end
+			total = total + 16;
+		}
+		//'total' variable now represents the minimum length for this chunk in order to write out all the primitives			
+		long newsize = this.getDataEndPosition() - this.getDataStartPosition();
+		if (newsize < total) {
+			newsize = total;
+		}
+		if (this.getParentChunk() != null) {
+			if (this.getParentChunk().usesPadByte()) {
+				newsize = KlangUtil.adjustForPadByte(newsize);
+			}
+		}
+		if (newsize > (this.getDataEndPosition() - this.getDataStartPosition())) {
+			long totnewsize = newsize + (this.getEndPosition() - this.getStartPosition()) - (this.getDataEndPosition() - this.getDataStartPosition());
+			//have to grow chunk
+			this.chunkIsAboutToChangeSize(totnewsize, raff);
+			int amt = total - (int)(this.getDataEndPosition() - this.getDataStartPosition());
+			byte[] newbyte = new byte[amt];
+			KlangUtil.insertIntoFile(newbyte, this.getParentFile().getFile(), this.getDataStartPosition());
+		}
+		raff.seek(this.getDataStartPosition());
+		//write values back to the file
+		counter = 0;
+		while (counter < end) {
+			if (counter == 6) {
+				try {
+					//write size
+					if (isBigEndian) {
+						Int2ByteUnsignedBE size = new Int2ByteUnsignedBE();
+						size.setValue((int)newsize - 18);
+						size.writeValueToFile(raff);
+					} else {
+						Int2ByteUnsignedLE size = new Int2ByteUnsignedLE();
+						size.setValue((int)newsize - 18);
+						size.writeValueToFile(raff);
+					}
+				} catch (Exception err) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)